File: clients/android/app/src/main/java/com/yeplist/app/data/remote/dto/CategoryDto.kt
package com.yeplist.app.data.remote.dto

import com.yeplist.app.data.local.entity.CategoryEntity

data class CategoryDto(
    val categoryId: Long,
    val name: String,
    val color: String?,
    val createdDate: String,
    val modifiedDate: String
) {
    fun toEntity(): CategoryEntity = CategoryEntity(
        categoryId = categoryId,
        name = name,
        color = color,
        createdDate = createdDate,
        modifiedDate = modifiedDate
    )
}

File: backend/src/ToDoList.Core/Dtos/CategoryDto.cs
using System;

namespace ToDoList.Core.Dtos
{
    public class CategoryDto
    {
        public long CategoryId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Color { get; set; }
        public int Version { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime ModifiedDate { get; set; }
        public DateTime ClientModifiedDate { get; set; }
    }
}
